pages/documents/DeveloperTools/FaaS/Developing/best-pratices.md

```diff
@@ -0,0 +1,241 @@
+  const secretClient = Toolbelt.SecretClient();
+
+  try {
+    const authUrl = 'www.mywebsite.com/auth';
+    const URL = 'www.mywebsite.com/request';
+
+    // Get clientId and clientSecret from secret storage
+    const client_id = await secretClient.readSecret('clientId');
+    const client_secret = await secretClient.readSecret('clientSecret');
+
+    // Obtain the token from the authorization server
+    const token = await hpptClient(authUrl, {
+        method: 'POST',
+        form: {
+            client_id: client_id.value,
+            client_secret: client_secret.value,
+            grant_type: 'client_credentials'
+        },
+        json: true,
+        simple: true,
+        resolveWithFullResponse: false
+    })
+
+    // Perform request with token (type may change)
+    const response = await httpClient(URL, {
+      method: "GET", 
+      headers: {
+        'Authorization': `Bearer ${token}`,
+      },
+      simple: false,
+      resolveWithFullResponse: true
+    });
+    
+    callback(null, token);
+  } catch (err) {
+    callback(err, null)
+  }
+}
+```
+
+### Usage of Environment Variables
+
+The Environment Variables are used to save configurations for a specific lambda. With this approach you can separate static variables from the code.
+
+<div class="important">
+  Do not save sensitive data in Environment Variables!
+</div>
+
+```javascript
+async function lambda(input, callback) {
+  const { Toolbelt } = require("lp-faas-toolbelt");
+  const httpClient = Toolbelt.HTTPClient();
+
+  try {
+    const response = await httpClient(process.env['CUSTOMER_URL'], {
+      method: "GET", 
+      headers: {}, 
+      simple: false,
+      resolveWithFullResponse: true
+    });
+      
+    const { statusCode, body } = response;
+    if(statusCode === 200) {
+      callback(null, process.env['SUCCESS_MESSAGE']);
+    } else {
+      callback(null, process.env['ERROR_MESSAGE']);
+    }
+  } catch (err) {
+    callback(err, null)
+  }
+}
+```
+
+See [Environment Variables](liveperson-functions-development-overview.html#environment-variables) for more information about it. 
+
+### Error handling in async functions
+
+If an async call has no correct error handling implemented the function will not end in the execution time and the error `Lambda Execution is taking too long` will be thrown.
+
+For [then-catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) promises it is necessary to have a `catch` at the end of a `then`.
+
+```javascript
+const myPromise =
+  (new Promise(myExecutorFunc))
+  .then(handleFulfilledA)
+  .then(handleFulfilledB)
+  .then(handleFulfilledC)
+  .catch(handleRejectedAny);
+```
+
+For [async-await](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Operators/await) a try-catch is necessary. 
+
+```javascript
+try {
+  await myPromise();
+} catch(error) {
+  // handle error
+} 
+```
+
+### Async/Await Functions
```

Comment:
move this part above the Error handling part. else it is strange that you already talk about the async/await functionality before introducing it. Now that I think of it... maybe you should this on top of the page, because you are already using `async/await` in some code examples earlier